pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DateTruncTransformFunction.java

```diff
@@ -75,6 +76,7 @@
  *   </li>
  * </ul>
  */
+@Deprecated
```

Comment:
For the deprecation, do we have enough query example coverages?

Reply:
How do I verify that? 
The date time scalar functions already cover all methods of DateTruncTransformFunction

Reply:
We should not deprecate this, but adding the support for 2 arguments as well.
Implementing a `TransformFunction` has much better performance than modeling it as scalar function.

Reply:
Ok.
So the issue is then the scalar functions which were imlpemented are not being used anywhere since all date_trunc calls go to the transform function.
I will add the 2 argument version in transform functions only then

Reply:
The scalar function will be used in post-aggregation, e.g. `date_trunc` on an aggregation result

Reply:
got it. Removed deprecation and added support for just 2 arguments in TransformFunction